providers/databricks/src/airflow/providers/databricks/hooks/databricks_base.py

```diff
@@ -141,6 +146,11 @@ def my_after_func(retry_state):
     def databricks_conn(self) -> Connection:
         return self.get_connection(self.databricks_conn_id)  # type: ignore[return-value]
 
+    async def a_databricks_conn(self) -> Connection:
+        if self._async_databricks_conn is None:
+            self._async_databricks_conn = await sync_to_async(self.get_connection)(self.databricks_conn_id)
```

Comment:
```suggestion
            if hasattr(self, "aget_connection"):
                self._async_databricks_conn = await self.aget_connection(self.databricks_conn_id)
            else:
                self._async_databricks_conn = await sync_to_async(self.get_connection)(self.databricks_conn_id)
```

Then we can use the async fn added in https://github.com/apache/airflow/pull/53831 if it's available.